src/monitoring/health_checks.py

```
@@ -0,0 +1,625 @@
+    def create_storage_health_check(self, storage_name: str, storage_path: str) -> Callable[[], ComponentHealth]:
+        """Create a health check function for storage."""
+        def check_storage_health() -> ComponentHealth:
+            try:
+                path = Path(storage_path)
+                
+                details = {
+                    'storage_name': storage_name,
+                    'storage_path': storage_path,
+                    'path_exists': path.exists(),
+                    'is_directory': path.is_dir() if path.exists() else False,
+                    'is_writable': os.access(path.parent if not path.exists() else path, os.W_OK)
```

Ruff (0.13.1)

Undefined name `os`

(F821)